apps/datapuller/src/shared/parser.ts

```diff
@@ -0,0 +1,191 @@
+import { ISectionItem } from "@repo/common";
+import { ClassSection } from "@repo/sis-api/classes";
+
+function getRequiredField<T>(
+  value: T | undefined,
+  fieldName: string,
+  defaultValue: T
+): T {
+  if (value === undefined || value === null) {
+    console.warn(`Missing required field: ${fieldName}`);
```

Comment:
Should probably use the `log` object. We can pass this down from the as a parameter (as opposed to use a global instance). 

This might be easiest to do by placing the `log` object into the `config` object, and then passing the `config` object downstream.